### DISCUSSION TITLE
why can't migrations be generated offline / without connection to the database?

### DISCUSSION THREAD

**Discussion by manan** [mikro-orm/mikro-orm#4226] [Q&A]
Given that there is a snapshot available, is there a reason why the cli tries to connect to the database when creating migrations instead of relying solely on the snapshot for the diff?

**Comment by haykerman** [comment]
I'm guessing because I might have snapshots disabled, so the source of truth would be the database.

**Reply by manan** [reply]
I have snapshots enabled and it still requires a connection to the db

**Reply by B4nan** [reply]
What version are you on?

**Reply by manan** [reply]
i'm on version 5.6.13

**Reply by manan** [reply]
it might be due to [this](https://github.com/mikro-orm/mikro-orm/blob/master/packages/cli/src/commands/MigrationCommandFactory.ts#L92) call here. `CLIHelper.getORM` invokes `MikroORM.init` that attempts to connect to the database. i tried setting `connect` to false in the config but that throws for some reason `Cannot read properties of undefined (reading 'raw')`

**Reply by manan** [reply]
@B4nan any thoughts?

**Reply by B4nan** [reply]
I think that in v6 we can just swap `init` with `initSync` and call it a day, so probably not worth spending time dealing with this in v5, I don't want to make the code more cluttered when there is a clean way to do it in v6.

> i tried setting connect to false in the config but that throws for some reason Cannot read properties of undefined (reading 'raw')

That's one of the things that are now resolved in v6 (#4166). DIsabling the connection means not having knex instance available in v5, so many things won't work.

**Reply by manan** [reply]
should i make a pr for this to get it in v6?

**Reply by B4nan** [reply]
Sure, PR welcome, but first ensure it actually works as expected.
